languages/ruby/exercises/concept/floating-point-numbers/.meta/design.md

```diff
@@ -0,0 +1,42 @@
+# Design
+
+## Goal
+
+The goal of this exercise is to teach the student the concept of floating-point numbers and introduce them to the `while` loop.
+
+## Learning objectives
+
+- Know about the [Float][float-class] class and some of its methods.
+- Know how to write a `while` loop.
+
+## Out of scope
+
+- Parsing floating-point types from strings.
+- Converting floating-point types to strings.
+- Using standard or custom format strings.
+
+## Concepts
+
+The Concepts this exercise unlocks are:
+
+- `floating-point-numbers`
+- `while-loops`: know how to write a `while` loop.
```

Comment:
```suggestion
- loops: know how to write a loop.
  - for loop
  - while loop
  - until loop
  - times loop
```
I almost don't think so, though.  We might be driving a specific solution, so perhaps.  But a for loop is not as common to use in Ruby (various reasons) and in general, it may be that we don't care so much which loop construct they use.

Reply:
I agree about the `until` loop, but I can't see a good way to use `for` or `times` to write the third method of this exercise

Reply:
Well, if you know the years, and you will compound the interest that many times, I think.  So if it is 5 years, then `5.times` seems reasonable.  But the math really seems like the best solution for that.

Reply:
But the years are what should be calculated:

### 3. Calculate the years before reaching the desired balance

Implement the `SavingsAccount.years_before_desired_balance` method to calculate the minimum number of years required to reach the desired balance:

```ruby
SavingsAccount.years_before_desired_balance(200.75, 214.88)
#=> 14
```

Reply:
I see. Yes. Sorry. OK. I'm good with this then. Thanks :)

Reply:
`loops` would be 4 different concepts. This exercise is only teaching one of those in the intro (while loops) so it's the other one that should be specified as a concept. If we want to introduce another concept here, it must go in the intro, after, and the example. So I don't think we have room for that as we only have one space in the example for a loop.